src/npcmove.cpp

```diff
@@ -3903,6 +3939,126 @@ void npc::reach_omt_destination()
     }
 }
 
+void npc::drop_job_products()
+{
+    assign_activity( activity_id( "ACT_TIDY_UP" ) );
+}
+
+void npc::stop_offscreen_job()
+{
+    offscreen_work_completed = 0;
+    offscreen_work_duration = cata::nullopt;
+    offscreen_work_started = cata::nullopt;
+    set_offscreen_job( NPC_OFFSCREEN_JOB_NULL );
+}
+
+void npc::return_to_base()
+{
+    tripoint where_is_home;
+    stop_offscreen_job();
+    if( !base_location ){
+        if( is_player_ally() ){
+            where_is_home = g->u.global_omt_location();
+        } else {
+            return;
+        }
+    } else {
+        where_is_home = *base_location;
+    }
+    if( global_omt_location() == where_is_home ){
+        talk_function::assign_guard( *this );
+        return;
+    }
+    travel_to_omt( where_is_home );
+}
+
+bool npc::forage_common( map &bay, const tripoint &pos )
+{
+    const harvest_id hid = bay.get_harvest( pos );
+    if( hid.is_null() || hid->empty() ) {
+        return true;
+    }
+    bool inventory_space = true;
+    const harvest_list &harvest = hid.obj();
+
+    int lev = get_skill_level( skill_survival );
+    for( const harvest_entry &entry : harvest ) {
+        float min_num = entry.base_num.first + lev * entry.scale_num.first;
+        float max_num = entry.base_num.second + lev * entry.scale_num.second;
+        int roll = std::min<int>( entry.max, round( rng_float( min_num, max_num ) ) );
+        if( roll >= 1 ) {
+            for( int i = 0; i < roll; i++ ) {
+                item harvest = item( entry.drop );
+                // we only want edibles, pine boughs just take up space.
+                if( harvest.typeId() == itype_id( "pine_bough" ) ){
```

Comment:
Perhaps later telling NPCs to pick up everything rather than just edibles could be an option later?

Reply:
yeah I'll add a smarter way for them to evaluate the harvest results.